### DISCUSSION TITLE
How do I double-indent `if` continuations?

### DISCUSSION THREAD

**Discussion by mid-kid** [uncrustify/uncrustify#4034] [Q&A] [answered]
My personal style does something like this:

```c
if (thing == VERY_LONG_ENUM_NAME_THING1 ||
        thing == VERY_LONG_ENUM_NAME_THING2) {
    do_some_thing();
}
if (some_long_func_call(param1, param2,
        param3, param4)) {
    do_some_thing();
}
```

I do this to distinguish the contents of the `if` statement from the rest of the block, without having to add an extra line in-between. I've been messing with the `indent_sparen_extra` option, and it doesn't seem to do anything. The default configuration always removes the extra indent.

**Comment by mid-kid** [comment, marked as answer]
Appaerently the `indent_func_call_param = true` option was overriding this and making it hard to get right, and `indent_sparen_extra` doesn't work unless `indent_continue` is nonzero
The following works, and does exactly what I want above:
```
indent_columns = 4
indent_continue = -4
indent_sparen_extra = 4
indent_func_call_param = false
```

**Comment by guy-maurel** [comment]
Please, provide an example which shows your problem.

**Reply by mid-kid** [reply]
The problem was not being able to get uncrustify to indent code in exactly the way shown in the OP, either always indenting 1 tab from the `if`, or modifying indentation depending on brace depth. I always indent `if` continuations with 2 tabs regardless of brace depth.
Most configurations (including the default configuration) would reindent it to something else, and the toggles to achieve this, as found in the accepted answer, weren't exactly obvious.
